src/core/providers/ProviderFactory.ts

```diff
@@ -27,87 +29,58 @@ export class ProviderFactory {
   }: IProviderFactory): Promise<DappProvider> {
     let createdProvider: IProvider | null = null;
     const address = getAddress();
+    const walletConnectConfig = walletConnectConfigSelector(getState());
 
     switch (type) {
       case ProviderTypeEnum.extension: {
-        const provider = await createExtensionProvider();
-        createdProvider = provider as unknown as IProvider;
-
-        createdProvider.getType = () => ProviderTypeEnum.extension;
+        const providerInstance = new ExtensionProviderStrategy(address);
```

Comment:
why are we passing the address since there is a fallback inside of the Strategy for it?